cmake/Modules/FindMagic_EP.cmake

```diff
@@ -0,0 +1,150 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# Finds the Magic library, installing with an ExternalProject as necessary.
+# This module defines:
+#   - libmagic_INCLUDE_DIR, directory containing headers
+#   - libmagic_LIBRARIES, the Magic library path
+#   - libmagic_FOUND, whether Magic has been found
+#   - The libmagic imported target
+
+# Include some common helper functions.
+include(TileDBCommon)
+
+# Search the path set during the superbuild for the EP.
+set(LIBMAGIC_PATHS ${TILEDB_EP_INSTALL_PREFIX})
+
+if(TILEDB_LIBMAGIC_EP_BUILT)
+  find_package(libmagic PATHS ${TILEDB_EP_INSTALL_PREFIX} ${TILEDB_DEPS_NO_DEFAULT_PATH})
+endif()
+
+if (TILEDB_LIBMAGIC_EP_BUILT)
+  find_path(libmagic_INCLUDE_DIR
+    NAMES magic.h
+    PATHS ${LIBMAGIC_PATHS}
+    PATH_SUFFIXES include
+    ${NO_DEFAULT_PATH}
+  )
+
+  if (NOT libmagic_INCLUDE_DIR)
+    find_path(libmagic_INCLUDE_DIR
+      NAMES file/file.h
+      PATHS ${LIBMAGIC_PATHS}
+      PATH_SUFFIXES include
+      ${NO_DEFAULT_PATH}
+    )
+  endif()
+
+  # Link statically if installed with the EP.
+  find_library(libmagic_LIBRARIES
+    libmagic
+    PATHS ${LIBMAGIC_PATHS}
+    PATH_SUFFIXES lib a
+    #${TILEDB_DEPS_NO_DEFAULT_PATH}
+    ${NO_DEFAULT_PATH}
+  )
+
+  find_library(libmagic_pcre2posix_LIBRARIES
+    pcre2-posix
+    PATHS ${LIBMAGIC_PATHS} ${TILEDB_EP_SOURCE_DIR}/ep_magic
+    PATH_SUFFIXES lib a
+    ${NO_DEFAULT_PATH}
+  )
+  find_library(libmagic_pcre2_8_LIBRARIES
+    pcre2-8
+    PATHS ${LIBMAGIC_PATHS} ${TILEDB_EP_SOURCE_DIR}/ep_magic
+    PATH_SUFFIXES lib a
+    #${TILEDB_DEPS_NO_DEFAULT_PATH}
+    ${NO_DEFAULT_PATH}
+  )
+
+  include(FindPackageHandleStandardArgs)
+  FIND_PACKAGE_HANDLE_STANDARD_ARGS(libmagic
+    REQUIRED_VARS libmagic_LIBRARIES libmagic_INCLUDE_DIR
+  )
+endif()
+
+# if not yet built add it as an external project
+if(NOT TILEDB_LIBMAGIC_EP_BUILT)
+  if (TILEDB_SUPERBUILD)
+    message(STATUS "Adding Magic as an external project")
+
+    if (WIN32)
+      set(CFLAGS_DEF "")
+    else()
+      set(CFLAGS_DEF "${CMAKE_C_FLAGS} -fPIC")
+      set(CXXFLAGS_DEF "${CMAKE_CXX_FLAGS} -fPIC")
+    endif()
+
+    # For both windows/nix using tiledb fork of file-windows from julian-r who has build a cmake version and patched for msvc++
+    # that was modified by tiledb to also build with cmake for nix
+    ExternalProject_Add(ep_magic
+      PREFIX "externals"
+      GIT_REPOSITORY "https://github.com/TileDB-Inc/file-windows.git"
+      GIT_TAG "nplat-cmake-install-support"
```

Comment:
Please use a SHA or tag here for reproducibility.

Reply:
changed

Reply:
This is merged so please make a follow-up PR for changes.